go.mod

```diff
@@ -17,8 +17,8 @@ require (
 	github.com/google/go-cmp v0.7.0
 	github.com/google/uuid v1.6.0
 	github.com/hashicorp/go-azure-helpers v0.72.0
-	github.com/hashicorp/go-azure-sdk/resource-manager v0.20250526.1224007
-	github.com/hashicorp/go-azure-sdk/sdk v0.20250526.1224007
+	github.com/hashicorp/go-azure-sdk/resource-manager v0.20250606.1221725
```

Comment:
Would you please split the PR into 3 PRs:
1. Update the `github.com/hashicorp/go-azure-sdk/resource-manager`
2. Update the `kubernetesconfiguration` from `2023-05-01` to `2024-11-01`.
3. Support for the new features.

Reply:
I opened this PR to update the SDK: https://github.com/hashicorp/terraform-provider-azurerm/pull/29896